src/Dialog/Dialog.tsx

```diff
@@ -241,60 +245,62 @@ const DefaultFooter: React.FC<DialogProps> = ({footerButtons}) => {
-    height = 'auto'
-  } = props
-  const dialogLabelId = uniqueId()
-  const dialogDescriptionId = uniqueId()
-  const defaultedProps = {...props, title, subtitle, role, dialogLabelId, dialogDescriptionId}
-
-  const dialogRef = useRef<HTMLDivElement>(null)
-  const combinedRef = useCombinedRefs(dialogRef, forwardedRef)
-  const backdropRef = useRef<HTMLDivElement>(null)
-  useFocusTrap({containerRef: dialogRef, restoreFocusOnCleanUp: true})
-
-  useOnEscapePress(
-    (event: KeyboardEvent) => {
-      onClose('escape')
-      event.preventDefault()
-    },
-    [onClose]
-  )
-
-  const header = (renderHeader ?? DefaultHeader)(defaultedProps)
-  const body = (renderBody ?? DefaultBody)(defaultedProps)
-  const footer = (renderFooter ?? DefaultFooter)(defaultedProps)
-
-  return (
-    <>
-      <Portal>
-        <Backdrop ref={backdropRef}>
-          <StyledDialog
-            width={width}
-            height={height}
-            ref={combinedRef}
-            role={role}
-            aria-labelledby={dialogLabelId}
-            aria-describedby={dialogDescriptionId}
-          >
-            {header}
-            {body}
-            {footer}
-          </StyledDialog>
-        </Backdrop>
-      </Portal>
-    </>
-  )
-})
+const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogProps & SxProp>>(
+  (props, forwardedRef) => {
+    const {
+      title = 'Dialog',
+      subtitle = '',
+      renderHeader,
+      renderBody,
+      renderFooter,
+      onClose,
+      role = 'dialog',
+      width = 'xlarge',
+      height = 'auto'
+    } = props
+    const dialogLabelId = uniqueId()
+    const dialogDescriptionId = uniqueId()
+    const defaultedProps = {...props, title, subtitle, role, dialogLabelId, dialogDescriptionId}
+
+    const dialogRef = useRef<HTMLDivElement>(null)
+    const combinedRef = useCombinedRefs(dialogRef, forwardedRef)
+    const backdropRef = useRef<HTMLDivElement>(null)
+    useFocusTrap({containerRef: dialogRef, restoreFocusOnCleanUp: true})
+
+    useOnEscapePress(
+      (event: KeyboardEvent) => {
+        onClose('escape')
+        event.preventDefault()
+      },
+      [onClose]
+    )
+
+    const header = (renderHeader ?? DefaultHeader)(defaultedProps)
+    const body = (renderBody ?? DefaultBody)(defaultedProps)
+    const footer = (renderFooter ?? DefaultFooter)(defaultedProps)
+
+    return (
+      <>
+        <Portal>
+          <Backdrop ref={backdropRef}>
+            <StyledDialog
+              width={width}
+              height={height}
+              ref={combinedRef}
+              role={role}
+              aria-labelledby={dialogLabelId}
+              aria-describedby={dialogDescriptionId}
+            >
+              {header}
+              {body}
+              {footer}
+            </StyledDialog>
+          </Backdrop>
+        </Portal>
+      </>
+    )
+  }
+)
 _Dialog.displayName = 'Dialog'
 
 const Header = styled(Box).attrs({as: 'header'})`
```

Comment:
Let's avoid extend `Box` now that it accepts all system props. That could create components that accept many more props than we expect.